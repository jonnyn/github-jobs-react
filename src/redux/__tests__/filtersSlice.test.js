import configureStore from 'redux-mock-store';
import { getDefaultMiddleware } from '@reduxjs/toolkit';
import filtersReducer, { INITIAL_STATE, setDescription, setLocation, setFullTime, setPage } from '../filtersSlice';

const middlewares = [...getDefaultMiddleware()];
const mockStore = configureStore(middlewares);

describe('Testing filters actions', () => {
    const store = mockStore(INITIAL_STATE);
    
    beforeEach(() => {
        store.clearActions();
      });

    it('should create an action to set description', () => {
        store.dispatch(setDescription('search keyword'));
        expect(store.getActions()).toMatchSnapshot();
    });

    it('should create an action to set location', () => {
        store.dispatch(setLocation('Vancouver'));
        expect(store.getActions()).toMatchSnapshot();
    });

    it('should create an action to set full_time', () => {
        store.dispatch(setFullTime(true));
        expect(store.getActions()).toMatchSnapshot();
    });

    it('should create an action to set page', () => {
        store.dispatch(setPage(2));
        expect(store.getActions()).toMatchSnapshot();
    });

});

describe('Testing filters reducers', () => {
    it('reduces correct states after setDescription action', () => {
        expect(filtersReducer(INITIAL_STATE, { type: 'filters/setDescription', payload: 'search keyword' })).toMatchSnapshot();
    });

    it('reduces correct states after setLocation action', () => {
        expect(filtersReducer(INITIAL_STATE, { type: 'filters/setLocation', payload: 'sf' })).toMatchSnapshot();
    });

    it('reduces correct states after setFullTime action', () => {
        expect(filtersReducer(INITIAL_STATE, { type: 'filters/setFullTime', payload: true })).toMatchSnapshot();
    });

    it('reduces correct states after setPage action', () => {
        expect(filtersReducer(INITIAL_STATE, { type: 'filters/setPage', payload: 3 })).toMatchSnapshot();
    });
});