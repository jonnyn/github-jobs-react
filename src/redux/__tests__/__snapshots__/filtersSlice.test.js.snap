// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing filters actions should create an action to set description 1`] = `
Array [
  Object {
    "payload": "search keyword",
    "type": "filters/setDescription",
  },
]
`;

exports[`Testing filters actions should create an action to set full_time 1`] = `
Array [
  Object {
    "payload": true,
    "type": "filters/setFullTime",
  },
]
`;

exports[`Testing filters actions should create an action to set location 1`] = `
Array [
  Object {
    "payload": "Vancouver",
    "type": "filters/setLocation",
  },
]
`;

exports[`Testing filters actions should create an action to set page 1`] = `
Array [
  Object {
    "payload": 2,
    "type": "filters/setPage",
  },
]
`;

exports[`Testing filters reducers reduces correct states after setDescription action 1`] = `
Object {
  "description": "search keyword",
  "fullTime": false,
  "location": null,
  "page": 1,
}
`;

exports[`Testing filters reducers reduces correct states after setFullTime action 1`] = `
Object {
  "description": null,
  "fullTime": true,
  "location": null,
  "page": 1,
}
`;

exports[`Testing filters reducers reduces correct states after setLocation action 1`] = `
Object {
  "description": null,
  "fullTime": false,
  "location": "sf",
  "page": 1,
}
`;

exports[`Testing filters reducers reduces correct states after setPage action 1`] = `
Object {
  "description": null,
  "fullTime": false,
  "location": null,
  "page": 3,
}
`;
